src/main/lrsql/auth/interceptor.clj

```diff
@@ -0,0 +1,63 @@
+(ns lrsql.auth.interceptor
+  (:require
+   [lrsql.input.auth :as auth-input]
+   [lrsql.ops.query.auth :as auth-q]
+   [lrsql.util :as util]
+   [next.jdbc :as jdbc]))
+
+(def holder (atom nil))
```

Comment:
I am failing to see the necessity of state here. Even looking at it i can't untangle why we would introduce it. We inject an interceptor, and then it modifies ctx, why are there atoms?

Reply:
state removed